### PR TITLE
feat: setup backend API structure, models & correct .gitignore

### PR DESCRIPTION
**Pull request by karansapolia**
Add configs, routes, env file and scripts to start and run
server.

targets: #7

## PR Checklist
Please check that your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/pesto-students/batch-9-Lumen/wiki/Commit-message-convention)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] GIFs/Videos/Images added (only for frontend changes)

## PR Type
What kind of change does this PR introduce? 

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Infrastructure changes
- [ ] Other... Please describe:

## Issue Number: 
#7 

## What is the new behavior?
Setups up server API structure with basic route
## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -3,7 +3,8 @@
     "browser": true,
     "es6": true,
     "node": true,
-    "jest": true
+    "jest": true,
+    "mongo": true
```
Comment (rishabh1403) [inline .eslintrc.json]: what does adding mongo do ?

File: server/index.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import app from './src/configs/server';
+
+const currentInstance = app();
```
Comment (rishabh1403) [inline server/index.js]: appInstance?

File: server/src/models/Playlist.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import mongoose from 'mongoose';
+import { songSchema } from './Song';
+
+const playlistSchema = new mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
```
Comment (rishabh1403) [inline server/src/models/Playlist.js]: playlist id ? to uniquely identify playlist

File: server/src/models/Playlist.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    user: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: 'User',
+      required: true,
+    },
+    songs: {
```
Comment (rishabh1403) [inline server/src/models/Playlist.js]: will you be adding songs in the same document ?

File: server/src/models/Song.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import mongoose from 'mongoose';
+
+const songSchema = new mongoose.Schema(
+  {
+    name: {
```
Comment (rishabh1403) [inline server/src/models/Song.js]: song id ?

File: server/src/models/User.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import mongoose from 'mongoose';
+
+const userSchema = new mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: true,
```
Comment (rishabh1403) [inline server/src/models/User.js]: id?

File: server/src/models/Playlist.js (hunk 3 above)
Comment (karansapolia) [reply server/src/models/Playlist.js]: mongo assigns a _id to every schema automatically. I'll use the toString() method of type ObjectId to convert _id to a string.

File: .eslintrc.json (hunk 1 above)
Comment (karansapolia) [reply .eslintrc.json]: It keeps ESLint from throwing 'not found' errors on mongo functions used in the global scope.

File: server/src/models/Playlist.js (hunk 4 above)
Comment (karansapolia) [reply server/src/models/Playlist.js]: This is currently creating new songs as a subdocument in playlist. Not the intended functionality. Will correct this to include ObjectId of song schema

File: server/src/models/Song.js (hunk 5 above)
Comment (karansapolia) [reply server/src/models/Song.js]: Same reasoning as [above](https://github.com/pesto-students/music-app-project/pull/9#discussion_r319467261). Thoughts?

File: server/src/models/User.js (hunk 6 above)
Comment (karansapolia) [reply server/src/models/User.js]: Same reasoning as [above](https://github.com/pesto-students/music-app-project/pull/9#discussion_r319467261). Thoughts?

**Comment by karansapolia** [comment]
@rishabh1403 this is a blocking PR. Should I merge for now and send fix PRs later if needed?

**Comment by karansapolia** [comment]
Merging to master. Please review my comments nonetheless.

**Pull request merged**